runtime/binding-grpc/src/main/java/io/aklivity/zilla/runtime/binding/grpc/internal/config/GrpcBindingConfig.java

```diff
@@ -107,13 +135,12 @@ public GrpcMethodResult resolveMethod(
             final CharSequence serviceName = serviceNameHeader != null ? serviceNameHeader : matcher.group(SERVICE_NAME);
             final String methodName = matcher.group(METHOD);
 
-            final GrpcMethodConfig method = options.protobufs.stream()
-                .map(p -> p.services.stream().filter(s -> s.service.equals(serviceName)).findFirst().orElse(null))
-                .filter(Objects::nonNull)
-                .map(s -> s.methods.stream().filter(m -> m.method.equals(methodName)).findFirst().orElse(null))
-                .filter(Objects::nonNull)
-                .findFirst()
-                .orElse(null);
+            GrpcMethodConfig method = resolveMethod(resolveCatalogs(), serviceName, methodName);
```

Comment:
If `resolveCatalogs()` is happening on every invocation of `resolveMethod` then we are repopulating the `protobufsBySchemaId` repeatedly, agree?

We don't usually pass `Stream` around, perhaps instead pass `Collection`?